src/Nemo.jl

```diff
@@ -353,28 +353,7 @@ function __init__()
    ccall((:flint_set_abort, libflint), Nothing,
          (Ptr{Nothing},), @cfunction(flint_abort, Nothing, ()))
 
-   # Check if were loaded from another package
-   # if VERSION < 1.7.*, only the "other" package will have the
-   # _tryrequire_from_serialized in the backtrace.
-   # if VERSION >= 1.8, also doing 'using Package' will have
-   # _tryrequire_from_serialized the backtrace.
-   #
-   # To still distinguish both scenarios, notice that
-   # 'using OtherPackage' will either have _tryrequire_from_serialized at least twice,
-   # or one with four arguments (hence five as the function name is the first argument)
-   # 'using Package' serialized will have a version with less arguments
-   bt = Base.process_backtrace(Base.backtrace())
-   filter!(sf -> sf[1].func === :_tryrequire_from_serialized, bt)
-   isinteractive_manual =
-      length(bt) == 0 || (length(bt) == 1 && length(only(bt)[1].linfo.specTypes.parameters) < 4)
-
-   # Respect the -q and --banner flag
-   allowbanner = Base.JLOptions().banner != 0
-
-   if allowbanner && isinteractive_manual && isinteractive() &&
-         !any(x -> x.name in ("Hecke", "Oscar", "Singular"), keys(Base.package_locks)) &&
-         get(ENV, "NEMO_PRINT_BANNER", "true") != "false"
-
+   if AbstractAlgebra.should_show_banner()
```

Comment:
```suggestion
   if AbstractAlgebra.should_show_banner() && get(ENV, "NEMO_PRINT_BANNER", "true") != "false"
```
I have no strong preference here. Thus, I let @fingolfin decide if to apply this patch here before merging.